### PR TITLE
Mashraf/transcribtion new reading mode ui

### PR DESCRIPTION
**Pull request by 22mahmoud**

### CODE REVIEW

File: src/pages/Transcribe/index.tsx [hunk 1]
```diff
@@ -290,6 +303,34 @@ class Transcribe extends React.Component<IProps, IState> {
         },
       },
       async () => {
+        if (!this.state.currentSurah) {
+          const [res1, res2] = await Promise.all([
+            fetch(`http://api.alquran.cloud/v1/surah/${ayahShape.chapter_id}`),
```
Comment (Ananasship) [inline src/pages/Transcribe/index.tsx]: we can add a TODO here saying that we'll refactor it to our own API in the future.

File: src/pages/Transcribe/index.tsx [hunk 2]
```diff
@@ -290,6 +303,34 @@ class Transcribe extends React.Component<IProps, IState> {
         },
       },
       async () => {
+        if (!this.state.currentSurah) {
+          const [res1, res2] = await Promise.all([
```
Comment (Ananasship) [inline src/pages/Transcribe/index.tsx]: We can add a comment explaining what res1 and res2 stand for. Might not be clear from the variable names.

File: src/pages/Transcribe/index.tsx [hunk 3]
```diff
@@ -290,6 +303,34 @@ class Transcribe extends React.Component<IProps, IState> {
         },
       },
       async () => {
+        if (!this.state.currentSurah) {
+          const [res1, res2] = await Promise.all([
+            fetch(`http://api.alquran.cloud/v1/surah/${ayahShape.chapter_id}`),
+            fetchSurah(ayahShape.chapter_id),
+          ]);
+          const surah = await res1.json();
+          const pages = surah.data.ayahs.reduce((acc, curr) => {
```
Comment (Ananasship) [inline src/pages/Transcribe/index.tsx]: what does `acc` stand for?

File: src/pages/Transcribe/index.tsx [hunk 4]
```diff
@@ -472,27 +511,40 @@ class Transcribe extends React.Component<IProps, IState> {
...
+                        {currentAyah.translations[0].text}
+                      </TranslationWrapper>
+                    )}
+                  </div>
+                )
               )}
             </div>
           </div>
-          <FooterWrapper>
+          <FooterWrapper isFullScreen={this.state.fullScreen}>
```
Comment (Ananasship) [inline src/pages/Transcribe/index.tsx]: love this! We can perhaps rename it to something like to `alignmentPosition={this.state.fullscreen ? 'left' : 'center'}` 

This way the naming wouldn't be tightly coupled to the full-screen mode and people can understand what this does w/o jumping into the FooterWrapper class.

File: src/pages/Transcribe/ReadingMode.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    } else {
+      this.setState({
+        rightPage: currentSurah[currentPage - 1],
+        leftPage: currentSurah[currentPage],
+      });
+    }
+  };
+  componentDidMount() {
+    const { currentAyah, currentSurah } = this.props;
+    if (currentAyah && currentSurah) {
```
[the remark this answers is not in the document]
Comment (22mahmoud) [reply src/pages/Transcribe/ReadingMode.tsx]: done

File: src/pages/Transcribe/ReadingMode.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        />
+      ))}
+    </PageWrapper>
+  );
+
+  render() {
+    const { currentSurah, currentAyah } = this.props;
+    const { leftPage, rightPage } = this.state;
+
+    if (!currentAyah) {
```
[the remark this answers is not in the document]
Comment (22mahmoud) [reply src/pages/Transcribe/ReadingMode.tsx]: No, this to make sure to render the component only if the currentAyah exist.

[the remark this answers is not in the document]
Comment (Ananasship) [reply src/pages/Transcribe/ReadingMode.tsx]: gotcha

File: src/pages/Transcribe/ReadingMode.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+  currentTranscribedIndex?: number;
+  currentSurah: ICurrentSurah;
+}
+
+interface IState {
+  leftPage: ISurahPage | null;
+  rightPage: ISurahPage | null;
+}
+
+const PagesWrapper = styled.div`
```
[the remark this answers is not in the document]
Comment (22mahmoud) [reply src/pages/Transcribe/ReadingMode.tsx]: done

**Pull request merged**